src/framework_detectors.rs

```diff
@@ -0,0 +1,173 @@
+use regex::Regex;
+
+use crate::testrun::Framework;
+
+fn gen_reg(s: &str) -> Regex {
+    Regex::new(format!(r"(?i){}(\W|$)", s).as_str()).unwrap()
+}
+
+fn apply_reg(rl: &[(Regex, Framework)], v: Vec<String>) -> Option<Framework> {
+    for (r, f) in rl {
+        for s in v.iter() {
+            if r.is_match(s) {
+                return Some(*f);
+            }
+        }
+    }
+    None
+}
+
+fn get_framework_names() -> [(Regex, Framework); 4] {
+    [
+        (gen_reg("pytest"), Framework::Pytest),
+        (gen_reg("jest"), Framework::Jest),
+        (gen_reg("vitest"), Framework::Vitest),
+        (gen_reg("phpunit"), Framework::PHPUnit),
+    ]
+}
+
+fn get_file_extensions() -> [(Regex, Framework); 2] {
+    [
+        (gen_reg(".py"), Framework::Pytest),
+        (gen_reg(".php"), Framework::PHPUnit),
+    ]
+}
```

Comment:
`Regex::new` is a notoriously slow operation, and you definitely don’t want to re-create those regexes on every function call.

I don’t think you need regexes at all for this. The `file_extension` check can just be a `.ends_with()` check for example.
The framework name might be a bit more complex, as you are matching for word boundaries. It might be a bit more complex there, using a `split_once`, and then checking each end of the split whether its a word boundary.
Might be quite a bit more code, but a ton more efficient than using regex for this.